src/flow/boykov_kolmogorov.jl

```diff
@@ -0,0 +1,208 @@
+"""
+Computes the max-flow/min-cut between source and target using
+Boykov-Kolmogorov algorithm.
+
+Returns the maximum flow in the network, the flow matrix and
+the partition {S,T} in the form of a vector of 1's and 2's.
+The partition vector may also contain 0's. These can be
+assigned any label (1 or 2), it is a user choice.
+
+For further details, please refer to the paper:
+
+BOYKOV, Y.; KOLMOGOROV, V., 2004. An Experimental Comparison of
+Min-Cut/Max-Flow Algorithms for Energy Minimization in Vision.
+
+Uses a default capacity of 1 when the capacity matrix isn\'t specified.
+
+Requires arguments:
+residual_graph::LightGraphs.DiGraph    # the input graph
+source::Int                            # the source vertex
+target::Int                            # the target vertex
+capacity_matrix::AbstractArray{T,2}    # edge flow capacities
+
+Author: Júlio Hoffimann Mendes (juliohm@stanford.edu)
+"""
+
+function boykov_kolmogorov_impl{T<:Number}(
+    residual_graph::LightGraphs.DiGraph,   # the input graph
+    source::Int,                           # the source vertex
+    target::Int,                           # the target vertex
+    capacity_matrix::AbstractArray{T,2}    # edge flow capacities
+    )
+
+    n = nv(residual_graph)
+
+    flow = 0
+    flow_matrix = zeros(T, n, n)
+
+    TREE = zeros(Int, n)
+    TREE[source] = 1
+    TREE[target] = 2
+
+    PARENT = zeros(Int, n)
+
+    A = [source,target]
+    O = Int[]
+
+    while true
+      # growth stage
+      path = find_path!(residual_graph, source, target, flow_matrix, capacity_matrix, PARENT, TREE, A)
+
+      isempty(path) && break
+
+      # augmentation stage
+      flow += augment!(path, flow_matrix, capacity_matrix, PARENT, TREE, O)
+
+      # adoption stage
+      adopt!(residual_graph, source, target, flow_matrix, capacity_matrix, PARENT, TREE, A, O)
+    end
+
+    return flow, flow_matrix, TREE
+end
+
+function find_path!{T<:Number}(
+    residual_graph::LightGraphs.DiGraph,   # the input graph
+    source::Int,                           # the source vertex
+    target::Int,                           # the target vertex
+    flow_matrix::AbstractArray{T,2},       # the current flow matrix
+    capacity_matrix::AbstractArray{T,2},   # edge flow capacities
+    PARENT::Vector{Int},                   # parent table
+    TREE::Vector{Int},                     # tree table
+    A::Vector{Int}                         # active set
+    )
+
+    tree_cap(p,q) = TREE[p] == 1 ? capacity_matrix[p,q] - flow_matrix[p,q] :
+                                   capacity_matrix[q,p] - flow_matrix[q,p]
+    while !isempty(A)
+      p = last(A)
+      for q in neighbors(residual_graph, p)
+        if tree_cap(p,q) > 0
+          if TREE[q] == 0
+            TREE[q] = TREE[p]
+            PARENT[q] = p
+            unshift!(A, q)
+          end
+          if TREE[q] ≠ 0 && TREE[q] ≠ TREE[p]
+            # p -> source
```

Comment:
Thanks! I love unicode, specially because in this case it follow the notation in the paper :blush: